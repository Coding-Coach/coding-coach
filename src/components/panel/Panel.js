import React from 'react';
import classNames from 'classnames';
import styles from './assets/panel.module.scss';

/**
 * Usage:
 *
 *   <Panel>
 *     <PanelHeader>Some title here</Panel.Header>
 *     <PanelContent>
 *       <img src="something.png" />
 *       <div>
 *         <p>We should be able to add anything as the content<.p>
 *       </div>
 *     </PanelContent>
 *     <PanelFooter>
 *        <small>you have x seconds to accept</small>
 *        <button>Click me!</button>
 *     </PanelFooter>
 *   </Panel>
 */
export function Panel({ classes, children }) {
  return (
    <div className={classNames(classes ? classes.panelRoot : null, styles.panel)}>{children}</div>
  );
}

export function PanelHeader({ children }) {
  return (
    <header className={styles.header}>
      <h2 className={styles.title}>{children}</h2>
    </header>
  );
}

export function PanelContent({ children }) {
  return <div className={styles.content}>{children}</div>;
}

export function PanelFooter({ children }) {
  return <footer className={styles.footer}>{children}</footer>;
}
