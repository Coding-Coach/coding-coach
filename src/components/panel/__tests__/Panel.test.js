import React from 'react';
import { render } from 'react-testing-library';
import { Panel, PanelContent } from '../Panel';

test('Renders basic panel', () => {
  const { container, queryByText } = render(<Panel>A basic Panel</Panel>);

  expect(container.firstChild).toMatchSnapshot();
  expect(queryByText('A basic Panel')).toBeDefined();
});
