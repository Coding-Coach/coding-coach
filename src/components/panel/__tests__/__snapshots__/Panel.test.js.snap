// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders basic panel 1`] = `
<div
  class="w-full h-full bg-white rounded-sm"
>
  A basic Panel
</div>
`;
